File: Others/medium/MoveElementToEnd.py
def moveElementToEnd(array, toMove):

    pointer = len(array) - 1

    for i in range(len(array) - 1, -1, -1):
        if array[i] == toMove:
            array[i], array[pointer] = array[pointer], array[i]
            pointer -= 1

    return array

File: Golang/medium/move-element-to-end.go
package main

func MoveElementToEnd(array []int, toMove int) []int {
	if len(array) == 0 {
		return array
	}

	lastElemPtr := len(array) - 1
	for lastElemPtr >= 0 && array[lastElemPtr] == toMove {
		lastElemPtr -= 1
	}

	if lastElemPtr <= 0 {
		return array
	}

	for idx := 0; idx < lastElemPtr; idx += 1 {
		if array[idx] == toMove {
			tmp := array[idx]
			array[idx] = array[lastElemPtr]
			array[lastElemPtr] = tmp
			for lastElemPtr >= 0 && array[lastElemPtr] == toMove {
				lastElemPtr -= 1
			}
			if lastElemPtr <= 0 {
				break
			}
		}
	}
	return array
}
